extensions-contrib/azure-extensions/src/main/java/io/druid/storage/azure/AzureDataSegmentPusher.java

```diff
@@ -72,7 +74,41 @@ public String getPathForHadoop(String dataSource)
   @Override
   public String getPathForHadoop()
   {
-    return null;
+    String hadoopPath = String.format(
+        Locale.ENGLISH,
+        "%s://%s@%s.blob.core.windows.net/",
+        "wasb",
+        config.getContainer(),
+        config.getAccount()
+    );
+
+    log.info("Using Azure blob storage Hadoop path: " + hadoopPath);
+
+    return hadoopPath;
+  }
+
+  @Override
+  public String getStorageDir(DataSegment dataSegment)
+  {
+
+    String seg = JOINER.join(
+        dataSegment.getDataSource(),
+        String.format(
+          Locale.ENGLISH,
+          "%s_%s",
+          // Use ISODateTimeFormat.basicDateTime() format, to avoid using colons in file path.
+          dataSegment.getInterval().getStart().toString(ISODateTimeFormat.basicDateTime()),
+          dataSegment.getInterval().getEnd().toString(ISODateTimeFormat.basicDateTime())
+        ),
+        dataSegment.getVersion(),
+        dataSegment.getShardSpec().getPartitionNum()
+    );
+
+    log.info("DataSegment: " + seg);
+
+    // Replace colons with underscores, since they are not supported through wasb:// prefix
+    return seg.replace(":", "_");
```

Comment:
Why doing this replace? From where colons could appear in `seg`?

Reply:
The dataSegment.getVersion() string is actually in the form of '2018-01-05T14:54:09.295Z' in my setup, coming from Druid I guess. I moved the conversion there. There is also a dataSegment.getBinaryVersion() field, I renamed any such occurrences to 'binaryVersion' to avoid confusion.